recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,257 @@
+            "queues",
+            "quicksight",
+            "ram",
+            "rds",
+            "redshift",
+            "recognition",
+            "resource_groups",
+            "robomaker"
+            "route53",
+            "route53domains",
+            "s3",
+            "sagemaker",
+            "sdb",
+            "serverlessrepo"
+            "servicecatalog",
+            "servicediscovery",
+            "shield",
+            "signer",
+            "sms",
+            "snowball",
+            "sns",
+            "sqs",
+            "ssm",
+            "states",
+            "storagegateway",
+            "sts",
+            "support",
+            "swf",
+            "text_to_speech",
+            "texttract",
+            "transcribe",
+            "transfer",
+            "translate",
+            "waf",
+            "workdocs",
+            "worklink",
+            "workmail",
+            "workspaces",
+            "xray"
+           )
+    options = merge_dicts_for_sdk({ x: [True, False] for x in sdks}, {
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "min_size": [True, False]
+        })
+    default_options = merge_dicts_for_sdk({ x: False for x in sdks}, {
+            "shared": False,
+            "fPIC": True,
+            "min_size": False
+        })
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("aws-c-common/0.4.25")
+        self.requires("aws-c-event-stream/0.1.5")
+        self.requires("zlib/1.2.11")
+        if self.settings.os != "Windows":
+            self.requires("libcurl/7.71.1")
+        if not self.settings.os in ["Windows", "Macos"]:
+            self.requires("openssl/1.1.1i")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+        shutil.rmtree(self._source_subfolder + "/code-generation")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        build_only = list([])
+        for sdk in self.sdks:
+            if getattr(self.options, sdk):
+                build_only.append(sdk)
+        self._cmake.definitions["BUILD_ONLY"] = ";".join(build_only)
+
+        self._cmake.definitions["BUILD_DEPS"] = False
+        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
+
+        self._cmake.definitions["MINIMIZE_SIZE"] = self.options.min_size
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```

Comment:
`BUILD_SHARED_LIBS` is automatically set by the conan cmake helper.
I think you should do:
```suggestion
        if self.settings.compiler == "Visual Studio":
            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in self.compiler.runtime
```
And test for compiler.runtime=MT and MD.

Reply:
It works!